src/sys/mman.rs

```diff
@@ -439,9 +437,10 @@ pub unsafe fn mmap<F: AsFd>(
 
     if ret == libc::MAP_FAILED {
         Err(Errno::last())
-    } else {
-        Ok(ret)
     }
+    else {
+        Ok(unsafe { NonNull::new_unchecked(ret) })
```

Comment:
Given that this is unsafe, do you think that we could add a comment stating it is safe, maybe something like:

> `mmap()` will never return a NULL pointer, it is safe to use `NonNull::new_unchecked()` here

Reply:
Added comment.